src/change_stream/options.rs

```diff
@@ -103,20 +108,46 @@ impl ChangeStreamOptions {
     }
 }
 
-#[rustfmt::skip]
 /// Describes the modes for configuring the
-/// [`ChangeStreamEvent::full_document`](crate::change_stream::event::ChangeStreamEvent::full_document)
-/// field.
+/// [`ChangeStreamEvent::full_document`](
+/// crate::change_stream::event::ChangeStreamEvent::full_document) field.
 #[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
 #[non_exhaustive]
 pub enum FullDocumentType {
-    #[rustfmt::skip]
-    /// The
-    /// [`ChangeStreamEvent::full_document`](crate::change_stream::event::ChangeStreamEvent::full_document)
-    /// field will be populated with a copy of the entire document that was updated.
-    #[serde(rename = "updateLookup")]
+    /// The field will be populated with a copy of the entire document that was updated.
     UpdateLookup,
 
+    /// The field will be populated for replace and update change events if the post-image for this
+    /// event is available.
+    WhenAvailable,
+
+    /// The same behavior as `WhenAvailable` except that an error is raised if the post-image is
+    /// not available.
+    Required,
+
+    /// User-defined other types for forward compatibility.
+    Other(String),
+}
+
+/// Describes the modes for configuring the
+/// [`ChangeStreamEvent::full_document_before_change`](
+/// crate::change_stream::event::ChangeStreamEvent::full_document_before_change) field.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub enum FullDocumentBeforeChangeType {
+    /// The field will be populated for replace, update, and delete change events if the pre-image
+    /// for this event is available.
+    WhenAvailable,
+
+    /// The same behavior as `WhenAvailable` except that an error is raised if the pre-image is
+    /// not available.
+    Required,
+
+    /// Do not send a value.
+    Off,
```

Comment:
this doesn't seem particularly useful but I see the spec has it so I guess we might as well too

Reply:
Yeah, I initially left it out but the spec tests use it.